WooCommerce/WooCommerceTests/System/WooAnalyticsTests.swift

```diff
@@ -90,6 +90,36 @@ class WooAnalyticsTests: XCTestCase {
         }
     }
 
+    /// Verifies an event with an error and properties is received by the AnalyticsProvider
+    ///
+    func testEventsWithPropertiesAndErrorReceived() {
+        let testError = NSError(domain: Constants.testErrorDomain, code: Constants.testErrorCode, userInfo: Constants.testErrorUserInfo)
+        analytics.track(.applicationOpened, properties: Constants.testProperty1, withError: testError)
+        XCTAssertEqual(testingProvider?.receivedEvents.count, 1)
+        XCTAssertEqual(testingProvider?.receivedProperties.count, 1)
+        XCTAssertEqual(testingProvider?.receivedEvents.first, WooAnalyticsStat.applicationOpened.rawValue)
```

Comment:
nit: we generally follow Given/When/Then in test cases so that it's more clear to differentiate code for each step:

```suggestion
        // Given
        let testError = NSError(domain: Constants.testErrorDomain, code: Constants.testErrorCode, userInfo: Constants.testErrorUserInfo)
        
        // When
        analytics.track(.applicationOpened, properties: Constants.testProperty1, withError: testError)
        
        // Then
        XCTAssertEqual(testingProvider?.receivedEvents.count, 1)
        XCTAssertEqual(testingProvider?.receivedProperties.count, 1)
        XCTAssertEqual(testingProvider?.receivedEvents.first, WooAnalyticsStat.applicationOpened.rawValue)
```